api/v1beta1/application_types.go

```diff
@@ -63,9 +71,9 @@ type Descriptor struct {
 
 // ApplicationSpec defines the specification for an Application.
 type ApplicationSpec struct {
-	// ComponentGroupKinds is a list of Kinds for Application's components (e.g. Deployments, Pods, Services, CRDs). It
+	// ComponentGVKs is a list of GVKs for Application's components (e.g. Deployments, Pods, Services, CRDs). It
 	// can be used in conjunction with the Application's Selector to list or watch the Applications components.
-	ComponentGroupKinds []metav1.GroupKind `json:"componentKinds,omitempty"`
+	ComponentGVKs []metav1.GroupVersionKind `json:"componentGVKs,omitempty"`
```

Comment:
isn't this a breaking change?

Reply:
also, "GVK" doesn't seem like a great external-facing API field name

Reply:
This is required for using controller-runtime v0.6.0, and we need to upgrade controller-runtime for K8s 1.21.

Reply:
> The version field is required for using controller-runtime v0.6.0+ (because of [kubernetes-sigs/controller-runtime#812](https://github.com/kubernetes-sigs/controller-runtime/pull/812)), and we need to upgrade controller-runtime for using v1 CRDs.

including version in the reference is not recommended... it means that any change in the available versions of the referenced types (e.g. apps/v1beta1 being removed in 1.16 in favor of apps/v1) would invalidate/require sweeping all references and updating them (e.g. to apps/v1). It would be better to resolve the preferred version of the referenced group/kind expressed in this API in the code that uses it and constructs the controller-runtime client

Reply:
xref https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#generic-object-reference for reference guidance